src/main/java/com/puppycrawl/tools/checkstyle/checks/AvoidEscapedUnicodeCharactersCheck.java

```diff
@@ -242,7 +242,10 @@ public void visitToken(DetailAST ast) {
      * @return true if literal has Unicode chars.
      */
     private static boolean hasUnicodeChar(String literal) {
-        return UNICODE_REGEXP.matcher(literal).find();
+        final Pattern escapedBackslashPattern = Pattern.compile("\\\\\\\\");
```

Comment:
please do compile and save it in final field of class, to improve performance a bit.